cmd/tree-walk-pool.go

```diff
@@ -121,21 +120,33 @@ func (t *TreeWalkPool) Set(params listParams, resultCh chan TreeWalkResult, endW
 				continue
 			}
 			// The first element is the oldest, so we only check that.
-			if v[0].added.Before(age) {
+			e := v[0]
+			if e.added.Before(age) {
 				oldest = k
+				age = e.added
 			}
 		}
 		// Invalidate and delete oldest.
-		if walks, ok := t.pool[oldest]; ok {
-			walk := walks[0]
-			walks[0] = treeWalk{} // clear references.
-			walks = walks[1:]
-			if len(walks) > 0 {
-				t.pool[params] = walks
+		if walks, ok := t.pool[oldest]; ok && len(walks) > 0 {
+			endCh := walks[0].endTimerCh
+			if len(walks) > 1 {
+				// Move walks forward
+				copy(walks, walks[1:])
+				walks = walks[:len(walks)-1]
+				if len(walks) > 0 {
```

Comment:
This if looks uneeded, it is also true here. (because it is in another condition where it makes sure that walks has at least two elements and we are removing only one element from it)

LGTM otherwise

Reply:
Yeah, just defensive programming.

Reply:
I can take it out, since it is pretty obviously not going to happen